pkg/monitor/agent/agent.go

```diff
@@ -78,12 +80,21 @@ type Agent struct {
 // goroutine and close all registered listeners.
 // Note that the perf buffer reader is started only when listeners are
 // connected.
-func NewAgent(ctx context.Context) *Agent {
+func NewAgent(ctx context.Context, optionalLimit ...int) *Agent {
+	var limit int
+	if len(optionalLimit) > 0 {
+		limit = optionalLimit[0]
+	}
+	var limiter *rate.Limiter
+	if limit > 0 {
```

Comment:
The `rate` pkg supports having a limiter with `rate.Inf` as its limit, so instead of making the limiter optional, we could just give it the infinite rate in the case where the user-passed limit is zero.

Reply:
Done